packages/gestalt/src/IconWithTooltip.jsdom.test.js

```diff
@@ -0,0 +1,55 @@
+// @flow
+import React from 'react';
+import { render, fireEvent } from '@testing-library/react';
+import IconWithTooltip from './IconWithTooltip.js';
+
+test('Icon with Tooltip and no href renders', () => {
+  const { getByRole, queryByRole } = render(
+    <IconWithTooltip
+      accessibilityLabel="informational-tooltip"
+      icon="circle-outline"
+      inline
+      tooltipText="Icon with Tooltip and Learn More link"
+    />
+  );
+
+  expect(getByRole('img')).toBeVisible();
+  fireEvent.mouseEnter(getByRole('img'));
+  expect(getByRole('tooltip')).toBeVisible();
+  expect(queryByRole('link')).toEqual(null);
+});
+
+test('Icon with Tooltip and href renders', () => {
+  const { getByRole, getAllByRole } = render(
+    <IconWithTooltip
+      accessibilityLabel="informational-tooltip"
+      icon="circle-outline"
+      inline
+      tooltipText="Icon with Tooltip and Learn More link"
+      href="https://www.pinterest.com"
+      hrefText="Learn More"
+    />
+  );
+
+  expect(getByRole('img')).toBeVisible();
+  fireEvent.mouseEnter(getByRole('img'));
+  expect(getByRole('tooltip')).toBeVisible();
+  expect(getAllByRole('link').length).toEqual(2);
+});
+
+test('Icon with Tooltip and href but missing href text renders', () => {
+  const { getByRole, queryByRole } = render(
+    <IconWithTooltip
+      accessibilityLabel="informational-tooltip"
+      icon="circle-outline"
+      inline
+      tooltipText="Icon with Tooltip and Learn More link"
+      href="https://www.pinterest.com"
+    />
+  );
+
+  expect(getByRole('img')).toBeVisible();
+  fireEvent.mouseEnter(getByRole('img'));
+  expect(getByRole('tooltip')).toBeVisible();
+  expect(queryByRole('link')).toEqual(null);
+});
```

Comment:
Please add test coverage for the other events handled by the component. Be sure to fire all of these and check for correct behavior: `fireEvent.mouseLeave(getByRole('img'))`, `fireEvent.mouseEnter(getByRole('tooltip'))`, `fireEvent.mouseLeave(getByRole('tooltip'))`

Reply:
Thanks @jennyscript - i've added a test for the full cycle.